Merge SDL3 upstream

capehill opened the pull request.

MBeijer commented on src/SDL_error.c:
```diff
@@ -125,6 +135,19 @@ SDL_SetError(SDL_PRINTF_FORMAT_STRING const char *fmt, ...)
+	int rc;
+	va_list ap;
+	va_start(ap, fmt);
+	rc = SDL_VSetError(fmt, ap);
```
I'm just curious, but hy have you done it like this?

BSzili replied:
If you mean the addition of the SDL_VSetError function, then it's to avoid variadic functions in the shared library interface.

capehill commented on src/cpuinfo/SDL_cpuinfo.c:
```diff
@@ -947,6 +966,9 @@ SDL_GetSystemRAM(void)
         }
 #endif
+#ifdef AMIGA
+        SDL_SystemRAM = AvailMem(MEMF_FAST | MEMF_TOTAL); // Exclude possible chip ram.
```
In MB: https://wiki.libsdl.org/SDL_GetSystemRAM

Could you use __MORPHOS__ Because AMIGA is compiled also on OS4.

capehill replied:
Seems better, but result is still in bytes, or?

BSzili replied:
Sorry, I missed the MB part.

capehill commented on src/file/SDL_rwops.c:
```diff
@@ -66,6 +66,30 @@
 #include "nacl_io/nacl_io.h"
 #endif
 
+#if defined(AMIGA)
```
Could you use __ MORPHOS __ ?

capehill commented on src/thread/morphos/sdl2-tc-userdata-replacement.diff.txt:
```diff
@@ -0,0 +1,95 @@
+Index: sdl2-main/src/thread/morphos/SDL_systhread.c
```
Is this file supposed to be here?

capehill commented on src/video/amiga/SDL_amigagl_wrapper.c:
```diff
@@ -0,0 +1,3125 @@
+/*
```
SDL2 uses ZLIB instead of LGPL

capehill commented on src/video/amiga/SDL_amigaopengl.c:
```diff
@@ -0,0 +1,263 @@
+/*
+    SDL - Simple DirectMedia Layer
```
zlib

capehill commented on src/video/amiga/SDL_amigaopengl.h:
```diff
@@ -0,0 +1,56 @@
+/*
```
zlib

MBeijer commented:
It looks like a lot of these changes could be useful for AmigaOS 3.x as well. :)

MBeijer commented:
@BSzili @BeWorld2018 
Since the official SDL finally moved over to Github, we decided to make a new fork from the official sources rather than the mirrors so we could continue pull changes from upstream. I then merged in our changes to the new fork, and then after doing that I synchronized your MorphOS fork into this branch: https://github.com/AmigaPorts/SDL-2.0/tree/MorphOS

So if you want to continue syncing with us, feel free to do a fork from there, all your latest changes should be in the branch I linked above, but it'd be nice if you can check if it still works for you. :)

BeWorld2018 commented:
Hi!
The last MorphOS version is now here : https://github.com/BeWorld2018/SDL and it's a fork of official SDL since few month :-)
I can change my repository again but..... i use link to my release version available (link... readme etc....)
I don't know how to do....

MBeijer commented:
Thanks for the info! I will merge in your changes to our MorphOS branch and
set up a new PR from your new fork :)

Kontrabant committed:
wayland: Throttle interactive user resize events

Excessive resize events generated during interactive window resizing can cause applications to lag, severely in some cases. Throttle interactive resize events to once per frame callback interval.

Kontrabant committed:
wayland: Check the WM capabilities before sending a minimize event

Wayland gives no feedback to windows regarding whether a minimize request was successful, so we would universally send the minimize event and assume that the request succeeded.

Check the window manager capabilities, if available, for whether window minimization is supported, and don't send the minimized event in the case that it is unsupported.

ccawley2011 committed:
testdrawchessboard: Allow using the standard render API

Commit:
Add macro guards for OpenGL <=1.3 function prototypes

- `#define SDL_OPENGL_1_NO_PROTOTYPES` to hide function prototypes
- `#define SDL_OPENGL_1_FUNCTION_TYPEDEFS` to add PFNGL*PROC function types

madebr committed:
Default functions are already declared by the first SDL_dynapi_procs.h include

madebr committed:
Refactor gendynapi.py with the final goal to make it re-usable

madebr committed:
Add parametrized build script

[ci skip]

DanielGibson committed:
Old env vars as fallback for SDL_VIDEO_DRIVER + SDL_AUDIO_DRIVER #11115

especially SDL_VIDEODRIVER is commonly used to use the native Wayland
backend, so I think it's a good idea to keep supporting the old name
instead of forcing users to find out that they now have to add an
underscore..
Not sure how popular SDL_AUDIODRIVER is, but with all the audio backends
that exist on Linux alone I'm sure some people use it to work around
sound issues.

Note: Doing this in the SDL_hints implementation instead of the
call-sites of SDL_GetHint(SDL_HINT_VIDEO_DRIVER) etc ensures that
1. Hint priorities work (env var overriding hint set by application with normal
   priority, but not when application used SDL_HINT_OVERRIDE)
2. SDL_ResetHint() (called by user code) respects the fallback
   environment variable

ccawley2011 committed:
Reduce the size of the SDL_blit_0 alpha code

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

ccawley2011 committed:
Allow for more fine tuning of Duff's device routines

ccawley2011 committed:
Remove leftover ARM SIMD code

ccawley2011 committed:
3DS: Only bundle resources with tests that need them

ccawley2011 committed:
Refactor Blit_RGB555_ARGB1555 to be more generic

maia-s committed:
Document main callbacks API thread safety

ccawley2011 committed:
3DS: Support simple message boxes

RustyMoyher committed:
SDL3: Fixed Cocoa_GL_CreateContext() not returning a context on success (#11181)

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

thatcosmonaut committed:
GPU Vulkan: Fix frame counter timing (#11189)

capehill committed:
Set size parameter in SDL_GetClipboardData()

Update size value in case where platform uses GetClipboardText().
This should fix clipboard_testClipboardDataFunctions on those platforms.

slouken committed:
Fixed SDL_oldnames.h to use the correct new names

slouken committed:
Added SDL_SetErrorV()

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Don't overwrite the SDL_IOFromFile() error in SDL_LoadBMP()

Fixes https://github.com/libsdl-org/SDL/issues/11188

Kontrabant committed:
wayland: Expose displays in a deterministic order, and attempt to better handle selecting a primary display.

Wayland can expose displays in any arbitrary order, and doesn't have the native concept of a primary display. However, there are games that presume that the first listed display is the primary, which can lead to problems if that output isn't necessarily the ideal one, as they may use that display to build a resolution list or as the default fullscreen output. This sorts displays by position, then attempts to find the primary display, first by querying the explicit ordering hint, then the GNOME DBus property, then tries to determine the 'best' display according to the criteria that is generally ideal for games and media playback.

The makes the list of displays deterministic, as long as the desktop configuration remains static, with a reasonably appropriate one prioritized as primary, even if there is no explicit way to retrieve the primary display. In the case where a user has a particularly esoteric physical display configuration, the new hint enables explicitly overriding the sorting and selection logic, allowing the entire display order to be customized, if necessary.

slouken committed:
Added SDL_GetDefaultLogOutputFunction()

Fixes https://github.com/libsdl-org/SDL/issues/10603
Closes https://github.com/libsdl-org/SDL/pull/11106

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Fixed Cohen-Sutherland out code computation for float line intersection

Fixes https://github.com/libsdl-org/SDL/issues/10866

slouken committed:
Check return value of SDL_small_alloc()

Fixes https://github.com/libsdl-org/SDL/issues/8959

madebr committed:
releases: don't add static library to MinGW binary release

icculus committed:
render: Add SDL_RenderDebugText().

Fixes #11201.

icculus committed:
test: remove testaudio-art.txt (this info is in test/LICENSE.txt now).

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

sezero committed:
SDL_windowsmodes.c: Fix MSVC builds against SDKs older than 10.0.17134.0

workaround the missing DISPLAYCONFIG_DEVICE_INFO_GET_SDR_WHITE_LEVEL enum value
and DISPLAYCONFIG_SDR_WHITE_LEVEL struct.

Fixes: https://github.com/libsdl-org/SDL/issues/11193
Closes:  https://github.com/libsdl-org/SDL/pull/11205

slouken committed:
Fixed uikit crash when GLES is disabled

Fixes https://github.com/libsdl-org/SDL/issues/11208

slouken committed:
Improved the documentation for SDL_SetWindowShape()

Fixes https://github.com/libsdl-org/SDL/issues/11199

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Added config.xcconfig to the Xcode build

thatcosmonaut committed:
GPU Vulkan: Fix depth-only framebuffer leak

thatcosmonaut committed:
GPU: Allow size queries on depth formats

slouken committed:
Added support for full range BT.709 YUV conversion

Also added simple colorspace testing to testyuv.

zturtleman committed:
Set error for NULL SDL_IOStream in SDL_LoadWAV_IO()

zturtleman committed:
Don't overwrite the SDL_IOFromFile() error in SDL_LoadFile()

zturtleman committed:
Don't overwrite the SDL_IOFromConstMem() error in testshape

zturtleman committed:
Make SDL_LoadBMP() use true instead of 1

icculus committed:
examples/game/01-snake: Update game logic in AppIterate, don't use a timer.

Reference Issue #11210.

slouken committed:
Updated documentation

Fixes https://github.com/libsdl-org/sdlwiki/issues/616

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Make sure stdio handles are in blocking mode

Standard I/O doesn't work well with non-blocking handles, so make sure any pipes are in blocking mode before launching child processes.

Fixes https://github.com/libsdl-org/SDL/issues/10998

isage committed:
PSVITA: enable posix fsops

TheSpydog committed:
GPU: Fix Metal sampler address mode order

madebr committed:
cmake: fix configuration with -DCMAKE_DISABLE_FIND_PACKAGE_PkgConfig=ON

madebr committed:
ci: update LoongArch toolchain to 2023.08.08

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

sezero committed:
CI, LoongArch64: disable clang-tidy

Reference issue: https://github.com/libsdl-org/SDL/issues/11219 .

Kontrabant committed:
wayland: Adjust primary display selection priority

Portrait displays are typically used as secondaries and almost certainly not what a user wants a game or movie initially becoming full screen on if a landscape display is available. Increase the priority of selecting a landscape display over a portrait display.

TheSpydog committed:
Improve support for private platforms (#11220)

slouken committed:
Rename GPU properties for consistency

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

madebr committed:
doc: explain build-scripts/create-release.py in the release checklist

kyle-sylvestre committed:
Report win32 last error string windows_file_open (#11224)

slouken committed:
Treat a button press location as an implicit mouse move

This fixes having the incorrect location for mouse button presses on Steam Deck under gamescope. This works around a bug where the mouse motion appears to come at the next button press, and comes from the xwayland-pointer device instead of the xwayland-relative-pointer device which is delivering mouse button events.

isage committed:
VITA: fix shader color format

slouken committed:
Removed fdatasync and gethostname definitions from platforms that don't have them

As verified by CMake builds in CI

1bsyl committed:
Fixed bug #10954: software renderer: examples/renderer/10-geometry missing a triangle
typo when changing return code from int to bool

slouken committed:
Make the name of the Steam Deck Controller consistent.

It will be named "Steam Deck Controller" regardless of whether you're running under Steam.

sezero committed:
cmake: fix checks for fdatasync() and gethostname()

maia-s committed:
Fix syntax of trailing doc comments

Sackzement committed:
Add `void` to `run_colorspace_test()` function prototype

slouken committed:
Added support for the Steam Virtual Gamepad on macOS Sequoia

slouken committed:
Added SDL_GetSandbox()

This adds support for detecting whether you're running in a sandbox on macOS

slouken committed:
Steam virtual gamepad info isn't available in the macOS sandbox

slouken committed:
Only use Steam Virtual Gamepads if we're running under Steam

Fixes https://github.com/libsdl-org/SDL/issues/11214
Fixes https://github.com/libsdl-org/SDL/issues/11215

sezero committed:
SDL_stdinc.h: provide a prototype for strdup for __clang_analyzer__ case

Because strdup is not ANSI but POSIX, and its prototype might be hidden..

... and re-enable clang-tidy in CI for LoongArch64.

Fixes:  https://github.com/libsdl-org/SDL/issues/11219.

slouken committed:
Re-added HAVE_FDATASYNC and HAVE_GETHOSTNAME for platforms that have them.

slouken committed:
SDL_ShouldIgnoreGamepad() should take product IDs rather than a guid.

This lets us early out when scanning for new hardware more quickly.

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

madebr committed:
ci: test Apple x64 on macos-13 runner

GitHub will remove macos-12 by December 3rd, 2024.

slouken committed:
Removed SDL_build_config_emscripten.h

The Emscripten build uses cmake, so we don't need the standalone config header.

Fixes https://github.com/libsdl-org/SDL/issues/11236

code-disaster committed:
Define SDL_DIALOG_DUMMY in SDL_build_config_xbox.h

sezero committed:
SDL_utils.c: restrict gethostname() usage to non-windows

Reference issue: https://github.com/libsdl-org/SDL/issues/11228.

isage committed:
VITA: fix yuv texture update

isage committed:
VITA: implement camera subsystem

slouken committed:
Treat clicking the right pad as a thumbstick click on a Steam Controller

slouken committed:
Requesting Bluetooth permission on Android is now asynchronous

This fixes an ANR at startup if gamepad input comes in while prompting for Bluetooth permissions.

Fixes https://github.com/libsdl-org/SDL/issues/6347

ebassi committed:
x11: Remove text-scaling-factor setting checks

The text-scaling-factor setting is not useful, now that we watch the
Xft/DPI and Gdk/WindowScalingFactor XSETTINGS keys; on the contrary, it
is problematic in mixed environments with both the KDE and GNOME
portals, as they end up stepping on each other's toes, and we end up
with the wrong scaling factor value.

Fixes: https://github.com/libsdl-org/SDL/issues/11142

slouken committed:
Use the correct return type for SDL_GetAndroidJNIEnv()

This prevents application code from having to cast to JNIEnv everywhere.

slouken committed:
Use SDL_GetAndroidJNIEnv() in hid.cpp

slouken committed:
Don't block in Android USB device open

We might be waiting a long time for a permissions dialog, so let that complete asynchronously. While that's happening, we'll remove the device from the device list so that when we get permission the application sees the device as newly available and can open it again.

Fixes https://github.com/libsdl-org/SDL/issues/6347

slouken committed:
Fixed crash if the Vulkan renderer changes swapchain image count

slouken committed:
Only free debug_char_texture_atlas if it has been allocated

Fixes https://github.com/libsdl-org/SDL/issues/11254

slouken committed:
Make sure that we quit if we get a terminating event

This covers the case where the application didn't handle the quit or terminating event, we still need the main loop to finish.

slouken committed:
If we get a newline character, treat it as SDLK_RETURN

Fixes https://github.com/libsdl-org/SDL/issues/10679

slime73 committed:
windows: fix race condition with exception-based thread naming.

sezero committed:
windows/SDL_systhread.c: fix permissions

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Added an OpenVR video driver (thanks @cnlohr!)

sezero committed:
video/openvr: Avoid type redefinition errors.

Also, let SDL handle stdint.h and stdbool.h.

Kontrabant committed:
kmsdrm: Support sorting displays via the priority hint

Use the connector name for displays and sort them according to priority, if the hint is set.

Kontrabant committed:
x11: Support sorting displays via the priority hint

Store the connector name for displays and use it for sorting them according to priority, if the hint is set.

slouken committed:
Renamed SDL_SANDBOX_UNKNOWN to SDL_SANDBOX_UNKNOWN_CONTAINER

maia-s committed:
Fix formatting

slouken committed:
Fixed assertion at quit when the debug text was the last texture used.

slouken committed:
openvr: added APIENTRY for OpenGL functions (thanks @cnlohr!)

huahua369 committed:
Fixed VULKAN_UpdateViewport judgment.

mausimus committed:
RI_KEY_BREAK indicates key is up

icculus committed:
audio: fix incorrect dynCall in Emscripten audio.

This only seemed to cause problems with `-s ASYNCIFY` on the build command
line, but it was incorrect in general.

Fixes #11258.

sezero committed:
add GLES3 headers from Khronos for OpenVR video driver.

rofrol committed:
typo: maxOS -> macOS

madebr committed:
docs: document copying shared SDL3 a bit more + document sanitizer/libc incompatibility

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

Kontrabant committed:
wayland: Remove kde_output_order_v1 support

This protocol is unstable and not intended for clients, and SDL now sorts the display list into a stable order and selects a primary display suitable for games on its own, so it isn't necessary.

Reverts e71e16950a5591913bd15bc8f4d04c01bf654b72

slouken committed:
Removed Blit2101010toN and BlitNto2101010

10-bit RGB formats are typically used with the PQ colorspace, which is handled by SDL_Blit_Slow.

Fixes https://github.com/libsdl-org/SDL/issues/11271

slouken committed:
Revert "Use the correct return type for SDL_GetAndroidJNIEnv()"

This reverts commit 2d3fa03d53b419bed2647cab2574ae27e9889586.

Different JDK implementations define JNIEnv differently for C++, so we can't provide the definition here.

Fixes:
/usr/lib/jvm/temurin-11-jdk-amd64/include/jni.h:195:17: error: conflicting declaration ‘typedef struct JNIEnv_ JNIEnv’
/home/runner/work/sdlwiki/sdlwiki/.github/../external/SDL3/include/SDL3/SDL_system.h:266:24: note: previous declaration as ‘typedef struct _JNIEnv JNIEnv’

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

Kontrabant committed:
Fix SDL_GetClosestFullscreenDisplayMode documentation

It returns false on failure, not NULL.

slouken committed:
Clamp the texture alpha and color modulation values to 0-255

rhett-lee committed:
SDL3: SDL_render_gl(render name: opengl) dose not support transparent on Linux/X11 (Bug #11273) (#11274)

SDL3: SDL_render_gl(render name: opengl) dose not support transparent on Linux/X11:

Create SDL window with SDL_WINDOW_TRANSPARENT flag;
Create "opengl" renderer for the window;
The window can't shown with errors:
X Error of failed request: BadMatch (invalid parameter attributes)
Major opcode of failed request: 130 (MIT-SHM)
Minor opcode of failed request: 3 (X_ShmPutImage)
Bug reason:
SDL_x11window.c:490 : bool X11_CreateWindow(SDL_VideoDevice *_this, SDL_Window *window, SDL_PropertiesID create_props)
SDL_x11window.c:566 : vinfo = X11_GL_GetVisual(_this, display, screen, transparent);[XVisualInfo *vinfo]
the X11_GL_GetVisual function returns a vinfo dose not support transparent.

Fix:
SDL_x11opengl.c:637 : XVisualInfo *X11_GL_GetVisual(SDL_VideoDevice *_this, Display *display, int screen, bool transparent)
X11_GL_GetVisual should returns a vinfo support transparent when transparent is true.

Sackzement committed:
SDL_RequestAndroidPermission() comment: Add link to android permissions website

Sackzement committed:
SDL_CreateRenderer() comment: Remove mention of flags

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

Sackzement committed:
Tweak android path functions comment tag \sa

slouken committed:
Added color scale info for the GPU renderer TODO item

slouken committed:
Fixed crashes handling D3D11/12 device lost in testsprite

You can test this using "dxcap -forcetdr"

slouken committed:
Added SDL_EVENT_RENDER_DEVICE_LOST

This is sent when the device is lost and can't be recovered.

Also fixed the vulkan renderer so it returns errors appropriately and will log and break if debug mode is enabled.

slouken committed:
Improved D3D12 handling of unrecoverable device loss

icculus committed:
render: GetRenderViewportSize shouldn't use scale, just logical presentation.

Fixes #11076.

slouken committed:
Don't treat mouse buttons as focus clicks if the window has mouse capture

This fixes the following sequence:
* Press mouse button down
* Alt-tab away from the window
* Alt-tab back to the window
* Release mouse button

Fixes https://github.com/libsdl-org/SDL/issues/7747

xordspar0 committed:
docs: Explain events for already plugged in controllers (#11286)

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Added logging for additional events

slouken committed:
Allow the OpenVR driver to be enabled on Windows

slouken committed:
Fixed memory leaks if OpenVR initialization fails

slouken committed:
Fixed memory leaks if Windows initialization fails

slouken committed:
Only set the OpenGL flag if the OpenVR driver is active

slouken committed:
Fixed warning C4456: declaration of 'e' hides previous local declaration

This also fixes incorrect interpretation of OPENVR_SetupJoystckBasedOnLoadedActionManifest() return value.

slouken committed:
Removed unused code

icculus committed:
docs: Updated README-emscripten.md.

This now mentions the main callbacks as a possible option, and removes the
part about prebuilt libraries. The latter can return when Emscripten packages
SDL3.

Fixes #11080.

anton-kl committed:
docs: fix window flags in README-migration.md

Use correct SDL macro instead of invalid "flags" string.
Also add a comment about preferred approach.

icculus committed:
x11: Remove the "safety net" error handler.

This is an X11 error catcher that lived for the entire life of the program,
which causes problems in itself, but it also tries to make calls into Xlib,
which would cause further panics.

It's goal was to reset video modes if the app was going down, but it often
failed to do so, and caused potentially avoidable crashes instead.

Fixes #7975.

icculus committed:
include: Updated docs on SDL_TouchFingerEvent.

Fixes #4159.

icculus committed:
wikiheaders: add (disabled) code to check for missing `\threadsafety` tags.

Reference Issue #7140.

madebr committed:
android: fix android.content.Context:getCacheDir JNI prototype

icculus committed:
docs: Fixed example code for SDL_qsort_r() and SDL_bsearch_r().

icculus committed:
docs: Add `\threadsafety` to some low-hanging fruit across a few headers.

Reference Issue #7140.

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

isage committed:
VITA: fix SDL_ShowMessageBox by using different memory type

icculus committed:
render: Corrected docs for SDL_SetRenderVSync().

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

icculus committed:
include: Fixed typo in SDL_CopyFile's documentation.

Fixes https://github.com/libsdl-org/sdlwiki/issues/623

slouken committed:
Fixed typo in the channel layout documentation

slouken committed:
Simplify get_channel_name() so it's easier for people to read and understand

Sackzement committed:
Fixed typo in SDL_clipboard.h

Sackzement committed:
Set version of new identifiers after 3.1.3 release to 3.1.6

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

mpalomas committed:
examples now using SDL_ALPHA_OPAQUE(_FLOAT) for opaque alpha value

Kontrabant committed:
wayland: Get the window content scale from the backend

The window content scale may change independently of the display it is on if scaling or accessibility features are involved, so query it directly from the backend instead of inferring it.

icculus committed:
fnsince: update `\since` policy for documentation.

Everything in SDL3 up to the ABI lock is reported as available since 3.1.3.
Everything else will be reported as since 3.2.0 (what will be the first
official release).

Also ran a Perl script over the headers to change everything to 3.1.3 that
wasn't an API function, since fnsince.pl can't manage those. If there's a
macro or datatype that has snuck in that needs to be 3.2.0 instead, we'll
have to manually fix it up, but it shouldn't be a big deal in any case.

Reference PR #11304.

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

thatcosmonaut committed:
GPU Vulkan: Use dedicated allocation for download buffers (#11298)

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

kyle-sylvestre committed:
Set error using function hresult

slouken committed:
testspriteminimal: only quit when escape is pressed

slouken committed:
evdev: fixed devices not being removed on disconnect

slouken committed:
evdev: correctly handle composite devices

These can be both mouse and keyboard and should be initialized and exposed to the application as both.

Sackzement committed:
Manually bump versions of non-function identifiers introduced after 3.1.3 to 3.2.0

Sackzement committed:
Bumped version of macro 'main' to 3.1.3

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

madebr committed:
cmake: fail configuring when no X11 or wayland could be found (#11310)

It's still possible to build SDL by configuring with -DSDL_NO_UNIX_DESKTOP_SANITY_CHECK=ON

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

icculus committed:
video: Turn several OpenGL attribute enums into integer typedefs.

Fixes #11308.

icculus committed:
video: Change capitalization on various OpenGL attribute typedefs.

icculus committed:
video: Change SDL_GLattr to SDL_GLAttr.

Susko3 committed:
Allow defining custom SDL_SINT64_C and SDL_UINT64_C macros (#11315)

Sackzement committed:
Allow redefinition of only one macro SDL_SINT64_C SDL_UINT64_C

icculus committed:
emscripten: scale mousewheel X coordinates correctly, not just Y coordinates.

Fixes #10454.

sonphantrung committed:
README-ios.md: Add info about UIApplicationSupportsIndirectInputEvents on iOS 17

Kontrabant committed:
Send drop complete events when the drop leaves the window on Cocoa, Wayland, and X11

This is already done on win32, however, other platforms were left in a state of limbo if a drop operation began, then never completed due to the drop leaving the window.

icculus committed:
docs: Added more '\threadsafety` tags.

Reference Issue #7140.

icculus committed:
audio: Make SDL_GetAudioStreamProperties() thread safe.

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

icculus committed:
cocoa: Make sure GL context destruction happens on the main thread.

Fixes #10900.

madebr committed:
android: add (disabled) android.permission.INTERNET to AndroidManifest.xml

It's required for SDL3_net

madebr committed:
snake: check allocated memory before use

madebr committed:
ci: add include headers of build directory to ci artifacts

[sdl-ci-artifacts]

slouken committed:
testcontroller: fixed incorrect usage of memcpy()

sezero committed:
video/openvr: minor clean-up:

- make all of function pointers static
- make EGLint context_attribs[] static
- comment out unused function pointer ov_wglGetCurrentContext
- remove unused SDL_DisplayMode openvr_dm_default
- move SDL_VideoDisplay openvr_vd_default to OPENVR_VideoInit()
  context and eliminate its C99 initializers
- replace atoi() calls with SDL_atoi()
- replace atof() calls with SDL_atof()

danginsburg committed:
Closes #10318 - implement Android prerotation in the Vulkan renderer

slouken committed:
build-release.py uses python3

slouken committed:
create-release.py uses python3

Sackzement committed:
SDL_EnumerateDirectory(): (posix) Fix return value when directory is invalid

gw3583 committed:
Update README-linux.md for openSUSE Tumbleweed dependencies

Sackzement committed:
SDL_Enumerate[Storage]Directory() comments: Use enum constant names instead of integers.

Kontrabant committed:
wayland: Refactor some shell surface related names

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

Sackzement committed:
SDL_CreateDirectory(): directory tree creation for absolute paths for non-Windows platforms

Sackzement committed:
SDL_SYS_RenamePath(): Fix error message.

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

flibitijibibo committed:
ci: Add Steam Linux Runtime (Sniper) task

TheSpydog committed:
Add GPU to iOS build config

TheSpydog committed:
metal: Add availability checks

expikr committed:
Splitcreen FPS example with multiple mouse and keyboard (#11317)

Susko3 committed:
Add examples to `SDL.sln` (#11255)

slouken committed:
Added instructions for creating Visual Studio projects for examples

slouken committed:
Added Visual Studio project for 02-woodeneye-008

Kontrabant committed:
wayland: Remove vestigial struct member

slouken committed:
Minor updates to woodeneye-008.c from template.c

TheSpydog committed:
Order by subsystem

d-musique committed:
metal: check success of device creation (#11367)

When macOS runs under a virtual environment, it is possible that
MTLCreateSystemDefaultDevice() does not succeed.
Not checking this failure results in a crash down the road.
This change allows to skip GPU renderer and use an adequate fallback.

Co-authored-by: D.musique <d-musique@users.noreply.github.com>

Dv-Senna committed:
SDL_vulkan.h: undefine VK_DEFINE_HANDLER and VK_DEFINE_NON_DISPATCHABLE_HANDLE

Allows including vulkan.hpp in user code in C++ before SDL_vulkan.h w/o errors.
Fixes:  https://github.com/libsdl-org/SDL/issues/11328 .

thatcosmonaut committed:
GPU D3D12: Fix indirect buffers not being refcounted

slouken committed:
Added 03-infinite-monkeys example game

This isn't really a game, more of an exercise of that age old question...

thatcosmonaut committed:
GPU D3D12: Fix depth texture sampling

Susko3 committed:
Rename variables for readability

Susko3 committed:
Specify C source file in example VC projects

thatcosmonaut committed:
GPU: Add SDL_CancelGPUCommandBuffer (#11316)

---------

Co-authored-by: Caleb Cornett <caleb.cornett@outlook.com>

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

madebr committed:
Allow in-tree build

slouken committed:
Autorelease keys should go down when pressed.

slouken committed:
Don't offset the view if there's no text input rect

Otherwise this will shift the entire view upwards, which probably isn't what you want.

slouken committed:
Mark generate.py as executable

Commit:
Supress Metal warning when compiling against macOS 10.11-10.13

Commit:
Add version info when building CI/CD for apple platforms

slouken committed:
sort_controllers.py: fixed parsing SDL_PRIVATE_GAMEPAD_DEFINITIONS

slouken committed:
Fixed the PS5 controller face buttons on Amazon Fire TV, Android 11

expikr committed:
typo in player spawnpos, mov drawcall out of loop

maia-s committed:
Add return to CHECK_VULKAN_ERROR_AND_RETURN macro

expikr committed:
Update SDL_mouse.c

expikr committed:
Update SDL_hints.h

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Minor cleanup

slouken committed:
Fixed return value setting state on combined Joy-Con controllers

sharkwouter committed:
PSP: Allow building tests in parallel

slouken committed:
Fixed the D-pad on wireless Xbox 360 controllers

sharkwouter committed:
PSP: Fix building examples in parallel

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Updated to version 3.1.6 for the preview release

slouken committed:
Updated to version 3.1.7 for development

SDLWikiBot committed:
Sync SDL3 wiki -> header

[ci skip]

slouken committed:
Set SDL_PROP_RENDERER_MAX_TEXTURE_SIZE_NUMBER for the GPU renderer

Fixes https://github.com/libsdl-org/SDL/issues/11393

slouken committed:
Updated the release checklist

This will help me remember to create the archives _after_ tagging the release.

madebr committed:
create-release.py: use head of git ref as default commit

icculus committed:
.wikiheader-options: Treat SDL int types (Uint64, etc) as API prefixes.

This makes any reference to them in the docs link to their wiki pages.
